### ISSUE TITLE
Chat-cli crashes on faulty text

### ISSUE THREAD

**Issue by rabsef-bicrym** [urbit/urbit#2125]
After a heavy day of chatting, I left for a while and came back to the error below. It appears dojo is still functional, I can reach other ships through hi, but I am not seeing any incoming chats nor do I see new chats I post to channels appear in the record after entering them in prompt

```
error in %chat-cli
%fact
! closing subscription
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.355 27].[1.355 49]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.355 17].[1.355 49]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.355 11].[1.355 49]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.354 9].[1.356 47]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.353 7].[1.361 36]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.348 7].[1.361 36]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.310 5].[1.362 7]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.309 5].[1.362 7]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.261 13].[1.261 23]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.261 9].[1.261 23]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.261 5].[1.293 21]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.254 5].[1.293 21]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.253 5].[1.293 21]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.112 13].[1.112 48]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.112 9].[1.112 48]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.112 5].[1.123 31]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.105 5].[1.123 31]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[1.104 5].[1.123 31]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[333 7].[333 45]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[333 3].[338 5]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[330 3].[338 5]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[329 3].[338 5]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[237 16].[237 70]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[234 3].[239 5]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[233 3].[239 5]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[128 25].[128 79]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[127 9].[129 11]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[122 7].[130 9]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[121 5].[131 17]>
! /~rabsef-bicrym/home/0/app/chat-cli/hoon:<[120 5].[131 17]>
! /~rabsef-bicrym/home/0/lib/verb/hoon:<[65 21].[65 44]>
! /~rabsef-bicrym/home/0/lib/verb/hoon:<[65 3].[66 15]>
! /~rabsef-bicrym/home/0/lib/verb/hoon:<[64 3].[66 15]>
! /~rabsef-bicrym/home/0/lib/verb/hoon:<[63 3].[66 15]>
! /~rabsef-bicrym/home/~2019.12.18..20.31.58..ab71/sys/vane/gall:<[1.246 9].[1.246 49]>
%chat-cli-kicked
```

**Comment by Fang-** [comment]
(In the future, use triple-` for multi-line codeblocks. (; )

This looks like it's crashing [here](https://github.com/urbit/urbit/blob/103e375417066e3a06319eef3299a8157f1f571b/pkg/arvo/app/chat-cli.hoon#L1355), which has been reported in the past, and is some weird text conversion edge case. I got it too. I suspect the offending message was in urbit-help, meaning everyone subscribed to it got affected too.

For some reason chat-cli [doesn't have `%kick` handling)[https://github.com/urbit/urbit/blob/103e375417066e3a06319eef3299a8157f1f571b/pkg/arvo/app/chat-cli.hoon#L125), so it doesn't automagically recover from this. I'll go and put that in, and try and see if I can get a hotfix OTA in to fix this.

In the past, I've been able to run `:chat-cli %connect` to fix the "no new messages in chat-cli" state. For some reason, this doesn't seem to work now. Will investigate.

**Comment by Fang-** [comment]
The "no `%kick` handling" part of this is resolved as per #2128.

**Comment by mocknotes** [comment]
cli spit out a bunch of error messages after the following message:

landscape:
![Screen Shot 2020-01-02 at 5 38 28 PM](https://user-images.githubusercontent.com/822317/71703030-72ea1600-2d87-11ea-8da4-b25730d6754c.png)

cli:
```
~lagrev-nocfep@ I'm trying to identify some runes that are in          ~01.35.27
                hoon.hoon but never described except perhaps in a
                pithy comment.
error in %chat-cli
%fact
! closing subscription
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.359 27].[1.359 49]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.359 17].[1.359 49]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.359 11].[1.359 49]>
! [%weird-text 664.415.896.517.041.070.978.924.807.425.738.741.010.076.875.809]
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.358 11].[1.359 49]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.357 9].[1.360 47]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.356 7].[1.365 36]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.351 7].[1.365 36]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.313 5].[1.366 7]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.312 5].[1.366 7]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.264 13].[1.264 23]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.264 9].[1.264 23]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.264 5].[1.296 21]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.257 5].[1.296 21]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.256 5].[1.296 21]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.115 13].[1.115 48]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.115 9].[1.115 48]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.115 5].[1.126 31]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.108 5].[1.126 31]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[1.107 5].[1.126 31]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[336 7].[336 45]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[336 3].[341 5]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[333 3].[341 5]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[332 3].[341 5]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[240 16].[240 70]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[237 3].[242 5]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[236 3].[242 5]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[128 25].[128 79]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[127 9].[129 11]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[122 7].[130 9]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[121 5].[131 17]>
! /~palnet-handel/home/0/app/chat-cli/hoon:<[120 5].[131 17]>
! /~palnet-handel/home/0/lib/verb/hoon:<[65 21].[65 44]>
! /~palnet-handel/home/0/lib/verb/hoon:<[65 3].[66 15]>
! /~palnet-handel/home/0/lib/verb/hoon:<[64 3].[66 15]>
! /~palnet-handel/home/0/lib/verb/hoon:<[63 3].[66 15]>
!   /~palnet-handel/home/~2019.12.18..21.34.42..ac9a/sys/vane/gall
    <[1.246 9].[1.246 49]>
subscribe failed from %chat-cli on wire /chat-store
watch-ack
/~zod/home/0/app/chat-store/hoon:<[52 5].[71 7]>
/~zod/home/0/app/chat-store/hoon:<[51 5].[71 7]>
/~zod/home/~2019.12.18..20.10.23..6846/sys/vane/gall:<[1.161 9].[1.161 37]>
```
As reported above cli no longer shows incoming messages, however I can still send and receive messages in landscape.

**Comment by Fang-** [comment]
Oops. #2147 should fix the `subscribe failed` part of this. If that had worked correctly, chat-cli would still crash, but then proceed to function normally.

For now, you can run `:chat-cli %connect` from dojo to reconnect it, so that it starts printing new messages again.

**Comment by belisarius222** [comment]
Just hit a 'weird-text' crash when joining ~/~dopzod/urbit-help.  I was still able to join successfully, though.  Not sure which text is implicated by the crash.

https://pastebin.com/nmz2W2Cy

**Comment by Fang-** [comment]
@belisarius222 this suggests you are running an older version of chat-cli. b35129e5 replaced the `%weird-text` crashes with a less noisy fallback message:

https://github.com/urbit/urbit/blob/18e396cf5383531f9ccdefc3090587f200268649/pkg/arvo/app/chat-cli.hoon#L1346-L1352

I saw you breached, and know we updated the pill very recently, so wonder how you got to be running outdated chat-cli...

**Comment by matildepark** [comment]
Closed via #2901.
